mediapipe/tasks/ios/vision/image_classifier/sources/MPPImageClassifier.mm

```diff
@@ -0,0 +1,228 @@
+                                                 roiAllowed:YES
+                                                      error:error];
+  if (!rect.has_value()) {
+    return nil;
+  }
+
+  Packet imagePacket = [MPPVisionPacketCreator createPacketWithMPPImage:image
+                                                            timestampMs:timestampMs
+                                                                  error:error];
+  if (imagePacket.IsEmpty()) {
+    return nil;
+  }
+
+  Packet normalizedRectPacket = [MPPVisionPacketCreator createPacketWithNormalizedRect:rect.value()
+                                                                           timestampMs:timestampMs];
+
+  PacketMap inputPacketMap = {{kImageInStreamName.cppString, imagePacket},
+                              {kNormRectName.cppString, normalizedRectPacket}};
+
+  std::optional<PacketMap> outputPacketMap =
+      [_visionTaskRunner processVideoFramePacketMap:inputPacketMap error:error];
+  if (!outputPacketMap.has_value()) {
+    return nil;
+  }
+
+  return
+      [MPPImageClassifierResult imageClassifierResultWithClassificationsPacket:
+                                    outputPacketMap.value()[kClassificationsStreamName.cppString]];
+}
+
+- (nullable MPPImageClassifierResult *)classifyVideoFrame:(MPPImage *)image
+                                              timestampMs:(NSInteger)timestampMs
+                                                    error:(NSError **)error {
+  return [self classifyVideoFrame:image
+                      timestampMs:timestampMs
+                 regionOfInterest:CGRectZero
+                            error:error];
+}
+
+- (BOOL)classifyAsyncImage:(MPPImage *)image
+               timestampMs:(NSInteger)timestampMs
+          regionOfInterest:(CGRect)roi
+                     error:(NSError **)error {
+  std::optional<NormalizedRect> rect =
+      [_visionTaskRunner normalizedRectFromRegionOfInterest:roi
+                                           imageOrientation:image.orientation
+                                                 roiAllowed:YES
+                                                      error:error];
+  if (!rect.has_value()) {
+    return NO;
+  }
+
+  Packet imagePacket = [MPPVisionPacketCreator createPacketWithMPPImage:image
+                                                            timestampMs:timestampMs
+                                                                  error:error];
+  if (imagePacket.IsEmpty()) {
+    return NO;
+  }
+
+  Packet normalizedRectPacket = [MPPVisionPacketCreator createPacketWithNormalizedRect:rect.value()
+                                                                           timestampMs:timestampMs];
+
+  PacketMap inputPacketMap = {{kImageInStreamName.cppString, imagePacket},
+                              {kNormRectName.cppString, normalizedRectPacket}};
+
+  return [_visionTaskRunner processLiveStreamPacketMap:inputPacketMap error:error];
```

Comment:
Would it be possible to create a helper for the common code in these three functions?

Reply:
Fixed.